doc/tutorials/feature_interaction_constraint.rst

```diff
@@ -0,0 +1,175 @@
+###############################
+Feature Interaction Constraints
+###############################
+
+The decision tree is a powerful tool to discover interaction among independent
+variables (features). Variables that appear together in a traversal path
+are interacting with one another, since the condition of a child node is
+predicated on the condition of the parent node. For example, the highlighted
+red path in the diagram below contains three variables: :math:`x_1`, :math:`x_7`,
+and :math:`x_{10}`, so the highlighted prediction (at the highlighted leaf node)
+is the product of interaction between :math:`x_1`, :math:`x_7`, and
+:math:`x_{10}`.
+
+.. plot::
+  :nofigs:
+
+  from graphviz import Source
+  source = r"""
+    digraph feature_interaction_illustration1 {
+      graph [fontname = "helvetica"];
+      node [fontname = "helvetica"];
+      edge [fontname = "helvetica"];
+      0 [label=<x<SUB><FONT POINT-SIZE="11">10</FONT></SUB> &lt; -1.5 ?>, shape=box, color=red, fontcolor=red];
+      1 [label=<x<SUB><FONT POINT-SIZE="11">2</FONT></SUB> &lt; 2 ?>, shape=box];
+      2 [label=<x<SUB><FONT POINT-SIZE="11">7</FONT></SUB> &lt; 0.3 ?>, shape=box, color=red, fontcolor=red];
+      3 [label="...", shape=none];
+      4 [label="...", shape=none];
+      5 [label=<x<SUB><FONT POINT-SIZE="11">1</FONT></SUB> &lt; 0.5 ?>, shape=box, color=red, fontcolor=red];
+      6 [label="...", shape=none];
+      7 [label="...", shape=none];
+      8 [label="Predict +1.3", color=red, fontcolor=red];
+      0 -> 1 [labeldistance=2.0, labelangle=45, headlabel="Yes/Missing           "];
+      0 -> 2 [labeldistance=2.0, labelangle=-45,
+              headlabel="No", color=red, fontcolor=red];
+      1 -> 3 [labeldistance=2.0, labelangle=45, headlabel="Yes"];
+      1 -> 4 [labeldistance=2.0, labelangle=-45, headlabel="             No/Missing"];
+      2 -> 5 [labeldistance=2.0, labelangle=-45, headlabel="Yes",
+              color=red, fontcolor=red];
+      2 -> 6 [labeldistance=2.0, labelangle=-45, headlabel="           No/Missing"];
+      5 -> 7;
+      5 -> 8 [color=red];
+    }
+  """
+  Source(source, format='png').render('../_static/feature_interaction_illustration1', view=False)
+  Source(source, format='svg').render('../_static/feature_interaction_illustration1', view=False)
+
+.. raw:: html
+
+  <p>
+  <img src="../_static/feature_interaction_illustration1.svg"
+    onerror="this.src='../_static/feature_interaction_illustration1.png'; this.onerror=null;">
+  </p>
+
+It is often the case that, when many variables are allowed to interact on
```

Comment:
When tree depth is larger than one, many variables interact on ..., and the resulting ...

Reply:
Fixed